src/track/beats.cpp

```diff
@@ -634,6 +637,20 @@ std::optional<BeatsPointer> Beats::tryTranslate(audio::FrameDiff_t offsetFrames)
             m_subVersion));
 }
 
+std::optional<BeatsPointer> Beats::tryTranslateBeats(double xBeats) const {
+    if (!m_markers.empty()) {
+        return std::nullopt;
+    }
+    const mixxx::audio::FrameDiff_t lastOffsetFrames =
+            xBeats * m_sampleRate.value() * 60.0 / m_lastMarkerBpm.value();
```

Comment:
The changes here seem very sensible, it would be nice if they were cemented in a unit test though. The test for the `Beats` class live in `src/test/beatgridtest.cpp`.